spec/struct.dd

```diff
@@ -566,6 +566,113 @@ $(GNAME Postblit):
+            static assert(!(__traits(hasMember, N2, "__xpostblit")));
+            static assert(!(__traits(hasMember, N2, "__postblit")));
+
+            // S has a defined postblit and a generated one
+            static assert((__traits(hasMember, S, "__xpostblit")));
+            static assert((__traits(hasMember, S, "__postblit")));
+
+            // B does not have a defined postblit, but has a generated one
+            static assert(!(__traits(hasMember, B, "__postblit")));
+            static assert((__traits(hasMember, B, "__xpostblit")));
+        }
+        ---
+
+        $(P Postblits are not overloadable.)
+        
+        $(P If a postblit is qualified with `immutable` or `shared` as in
+        $(D this(this) immutable;) or $(D this(this) shared;), then the code
+        is ill-formed. Historically, such definitions passed compilation but
+        could not be invoked. A deprecation diagnostic is issued by the front-end
+        of dmd 2.080 or later, see $(BUGZILLA 18417). Example:)
+
+        ---
+        struct S1
+        {
+            // ill-formed/not callable, no diagnostic before dmd 2.080
+            this(this) immutable {}
+        }
+        struct S2
+        {
+            // ill-formed/not callable, no diagnostic before dmd 2.080
+            this(this) shared {}
+        }
+        ---
+
+        $(P If a postblit is qualified with `const` as in $(D this(this) const;),
+        the code is ill-formed. A deprecation diagnostic is issued by the
+        front-end of dmd 2.080 or later, see $(BUGZILLA 18417). Prior versions
+        inserted appropriate calls to the postblit, but the postblit cannot modify
+        the object because it regards it as `const`; hence, `const` postblits
+        were of limited usefulness.)
+
+        ---
+        struct U
+        {
+            int n;
+            // ill-formed, no diagnostic before dmd 2.080
+            this(this) const
+            {
+                ++n;       // Error: cannot modify this.n in const function
+            }
+        }
+
+        void main()
+        {
+            S a;
+            auto a2 = a;   // Error: immutable method test.S.__postblit is not callable using a mutable object
+            immutable S a3;
+            auto a4 = a3;  // Error: immutable method test.S.__postblit is not callable using a mutable object
+
+            U u;
+            auto u2 = u;
+        }
+        ---
+
+        $(P An unqualified postblit will get called even if the
+        struct is instantiated as `immutable` or `const`:)
+
+        ---
+        struct S
+        {
+            int n;
+            this(this) { ++n; }
+        }
+
+        void main()
+        {
+            immutable S a;
+            auto a2 = a;
+            import std.stdio: writeln;
+            writeln(a2.n);     // prints 1
+        }
+        ---
+        
+        $(P This allows incorrect code to compile, refer to $(BUGZILLA 18357) for examples.)
```

Comment:
This doesn't work. The spec should be self-containing as Bugzilla might be offline or it simply might be print as ebook or PDF.